homeassistant/components/venstar/climate.py

```diff
@@ -258,32 +258,28 @@ def _set_operation_mode(self, operation_mode: HVACMode):
             _LOGGER.error("Failed to change the operation mode")
         return success
 
-    def set_temperature(self, **kwargs):
+    def set_temperature(self, **kwargs: Any) -> None:
         """Set a new target temperature."""
         set_temp = True
-        operation_mode = kwargs.get(ATTR_HVAC_MODE)
-        temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
-        temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
-        temperature = kwargs.get(ATTR_TEMPERATURE)
-
-        if operation_mode and self._mode_map.get(operation_mode) != self._client.mode:
+        operation_mode: HVACMode | None = kwargs.get(ATTR_HVAC_MODE)
+        temp_low: float | None = kwargs.get(ATTR_TARGET_TEMP_LOW)
+        temp_high: float | None = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+        temperature: float | None = kwargs.get(ATTR_TEMPERATURE)
+
+        client_mode = self._client.mode
+        if (
+            operation_mode
+            and (new_mode := self._mode_map.get(operation_mode)) != client_mode
+        ):
             set_temp = self._set_operation_mode(operation_mode)
+            client_mode = new_mode
```

Comment:
Store the updated value, so we don't need to use `self._mode_map.get` over and over below